pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -435,13 +436,18 @@ func wellKnownModule(name string) bool {
 }
 
 // EnableServiceMonitor add a servicemonitor that allows prometheus to scrape from the monitoring endpoint of the cluster
-func (c *Cluster) EnableServiceMonitor(activeDaemon string) error {
+func (c *Cluster) EnableServiceMonitor(activeDaemon string, labels map[string]string) error {
 	serviceMonitor, err := k8sutil.GetServiceMonitor(path.Join(monitoringPath, serviceMonitorFile))
 	if err != nil {
 		return errors.Wrap(err, "service monitor could not be enabled")
 	}
 	serviceMonitor.SetName(AppName)
 	serviceMonitor.SetNamespace(c.clusterInfo.Namespace)
+	err = mergo.Merge(&serviceMonitor.ObjectMeta.Labels, labels, mergo.WithOverride)
```

Comment:
How about following the same pattern as other labels? For example, the mons apply the custom labels like this:
```
cephv1.GetMonLabels(c.spec.Labels).ApplyToObjectMeta(&d.ObjectMeta)
```
We certainly might want to convert to mergo, but that seems better for a separate PR inside the `ApplyToObjectMeta()` method.

Reply:
Made the change. But, `ApplyToObjectMeta()` does not merge existing keys, it only adds new ones.
We'll eventually need to add a way to override existing labels.